slack/resource_conversation.go

```diff
@@ -141,6 +159,54 @@ func resourceSlackConversationCreate(ctx context.Context, d *schema.ResourceData
 	return resourceSlackConversationRead(ctx, d, m)
 }
 
+func findExistingChannel(ctx context.Context, client *slack.Client, name string, isPrivate bool) (*slack.Channel, error) {
+	// find the existing channel. Sadly, there is no non-admin API to search by name,
+	// so we must search through ALL the channels
+	tflog.Info(ctx, "Looking for channel %s", name)
+	paginationComplete := false
+	cursor := ""       // initial empty cursor to begin at start of list
+	var types []string // default value with empty list is "public_channel"
+	if isPrivate {
+		types = append(types, "private_channel")
+	}
+	for !paginationComplete {
+		channels, nextCursor, err := client.GetConversationsContext(ctx, &slack.GetConversationsParameters{
+			Cursor: cursor,
+			Limit:  200, // 100 is default, docs recommend no more than 200, but 1000 is the max
+			Types:  types,
+		})
+		tflog.Debug(ctx, "got %d channels, nextCursor %s, err: %v", len(channels), nextCursor, err)
+		if err != nil {
+			if rateLimitedError, ok := err.(*slack.RateLimitedError); ok {
+				tflog.Warn(ctx, "rate limited for %f seconds", rateLimitedError.RetryAfter.Seconds())
+				select {
+				case <-ctx.Done():
+					return nil, fmt.Errorf("canceled during pagination: %s", ctx.Err())
+				case <-time.After(rateLimitedError.RetryAfter):
+					tflog.Debug(ctx, "done sleeping after rate limited")
+				}
```

Comment:
This is similarly implemented to the internals of the paginated GetUsersContext in the slack-go library that this provider is built upon.
https://github.com/slack-go/slack/blob/34c7fd379e3526317abcc057ef7baa686cb0a42c/users.go#L379